eng/pipelines/templates/stages/archetype-ios-release.yml

```diff
@@ -66,17 +66,32 @@ stages:
 
           - deployment: PublishPackageToSwiftPackageManagerMirrors
             displayName: "Publish to SPM Mirrors"
-            condition: and(succeeded(), ne(variables['Skip.PublishPackage'], 'true'))
+            condition: and(succeeded(), ne(variables['Skip.PublishPackage'], 'true'), ne('${{artifact.skipSwiftPackageManager}}', true))
             environment: cocoapodstrunk
             dependsOn: TagRepository
             variables:
               - template: ../variables/globals.yml
             pool:
-              name: Azure Pipelines
-              vmImage: $(OSVmImage)
+              name: azsdk-pool-mms-ubuntu-1804-general
+              vmImage: MMSUbuntu18.04
             strategy:
               runOnce:
                 deploy:
                   steps:
-                    - script: |
-                        echo "Pretend we are publishing to Swift Package Manager mirrors"
+                    - checkout: self
+                    - checkout: SwiftPM-${{artifact.name}}
+                      persistCredentials: true
+                    - pwsh: |
+                        $podspec = Get-Content -Raw -Path $(Pipeline.Workspace)/s/azure-sdk-for-ios/sdk/${{parameters.ServiceDirectory}}/${{artifact.name}}/${{artifact.name}}.podspec.json | ConvertFrom-Json
+                        $version = $podspec.version
+                        Push-Location -Path $(Pipeline.Workspace)/s/SwiftPM-${{artifact.name}}
+                        git checkout --orphan run-$(Build.BuildID)-temp-branch
+                        git reset --hard
```

Comment:
We should explicitly put a ref to reset here instead of relying on defaults.

Reply:
The preceding command basically creates a new ref pointing to an empty state. The reset is just an easy way of getting rid of the files left over from master/main.